### PR TITLE
Philip/wal 9 sell p1 buy from your fiat balance

### PR DESCRIPTION
**Pull request by plondon**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/actions.ts [hunk 1]
```diff
@@ -373,9 +373,13 @@ export const initializeBillingAddress = () => ({
   type: AT.INITIALIZE_BILLING_ADDRESS
 })
 
-export const initializeCheckout = (orderType: 'BUY' | 'SELL') => ({
+export const initializeCheckout = (
+  orderType: 'BUY' | 'SELL',
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/actions.ts:377]: maybe to make type of this like `type OrderType = 'BUY' | 'SELL'`

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts [hunk 2]
```diff
@@ -12,8 +14,23 @@ export const getCryptoCurrency = (state: RootState) =>
...
-  state.components.simpleBuy.defaultMethod
+export const getDefaultPaymentMethod = (state: RootState) => {
+  const ordersR = getSBOrders(state)
+  const sbMethodsR = getSBPaymentMethods(state)
+
+  const transform = (
+    orders: ExtractSuccess<typeof ordersR>,
+    sbMethods: ExtractSuccess<typeof sbMethodsR>
+  ) => {
+    const { paymentType: type, inputCurrency } = orders[0]
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts:25]: is there a chance to have orders as `[]` ?

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/CryptoSelection/CryptoSelector/CryptoItem/selectors.ts [hunk 3]
```diff
@@ -10,9 +10,16 @@ export const getData = (state: RootState, ownProps: OwnProps) => {
...
-  return lift((rates: ExtractSuccess<typeof ratesR>) => ({
-    fiatCurrency,
-    rates
-  }))(ratesR)
+  return lift(
+    (
+      rates: ExtractSuccess<typeof ratesR>,
+      supportedCoins: ExtractSuccess<typeof supportedCoinsR>
+    ) => ({
+      fiatCurrency,
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/CryptoSelection/CryptoSelector/CryptoItem/selectors.ts:20]: do we need `fiatCurrency` here?

Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/CryptoSelection/CryptoSelector/CryptoItem/selectors.ts:20]: oh we do, lift add only remote, gotcha

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx [hunk 4]
```diff
@@ -112,14 +106,21 @@ const getIcon = (value: SBPaymentMethodType): ReactElement => {
         />
       )
     case 'FUNDS':
-      return <></>
+      return (
+        <Icon
+          size='32px'
+          color='fiat'
+          name={value.currency.toLowerCase() as keyof IcoMoonType}
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx:113]: 👏

**Comment by milan-bc** [comment]
@plondon great work so far, I notice few things

image


in figma we have such thing centred also I didn't found this one

image

is that going to be added later?

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts (hunk 2 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts:25]: Nice catch.

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx (hunk 4 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx:113]: Yeah it was IcomoonType not DefaultTheme, forgot DefaultTheme is for colors :)

**Comment by plondon** [comment]
Yup this was moved to p2

**Pull request merged**